File: php/www/experiment/oo/java/Person.java
// Person.java

public class Person {
	protected String firstName;
	protected String middleName;
	protected String lastName;
	protected String fullName;

	public Person(String firstName, String middleName, String lastName){
		this.firstName = firstName;
		this.middleName = middleName;
		this.lastName = lastName;
		setFullName();
	}

	protected void setFullName(){
		System.out.println("Used setFullName() from Person");
		this.fullName = this.firstName + " " + this.lastName;
	}

	public String getFullName(){
		return this.fullName;
	}

}

File: blogExamples/otherLanguages/js/HB/Person.js
HB.accounts = HB.accounts || {};

HB.accounts.Person = function(personContainer, firstNameSelector, lastNameSelector){
	this.$personContainer = $(personContainer);
	this.$firstName = $(firstNameSelector, this.$personContainer);

	this.$lastName = $(lastNameSelector, this.$personContainer);
	this.$fullName = HB.accounts.Person.displayFullNameControl(this.$personContainer);

	this.$personContainer.on("keyup", [firstNameSelector,lastNameSelector], $.proxy(HB.accounts.Person.keyUpHandler, this));
};

HB.accounts.Person.displayFullNameControl = function(el){
	var $fullName = $('<input type="text" name="fullName" value="">');
	el.append("Full name: ");
	$fullName.appendTo(el);
	el.append("<br>");
	return $fullName;
};

HB.accounts.Person.keyUpHandler = function(){
	var fullName = this.getFullName();
	this.setFullName(fullName);
};

HB.accounts.Person.prototype.setFullName = function(fullName){
	this.$fullName.val(fullName);
}

HB.accounts.Person.prototype.getFullName = function(){
	var fullName = this.$firstName.val() + " " + this.$lastName.val();
	return fullName;
};
